lib/service/database.js

```diff
@@ -87,7 +87,9 @@ DatabaseService.prototype.getMetadataFromAffectedTables = function(node, skip, c
         var defaultLastUpdatedTime = new Date('1970-01-01T00:00:00.000Z');
         var affectedTables = [];
         var lastUpdatedTimes = [];
-        var rows = resultSet.rows || [defaultLastUpdatedTime, null, null];
+        var rows = (Array.isArray(resultSet.rows) && resultSet.rows.length > 0) ?
+            resultSet.rows :
+            [defaultLastUpdatedTime, null, null];
```

Comment:
If we use an empty array we are falling in the same bug that we want to fix, don't we?